src/Nethermind/Nethermind.Core/Transaction.cs

```diff
@@ -72,40 +75,49 @@ void GenerateHash()
             }
             set
             {
-                if (_preHash.Length > 0)
+                lock (this)
                 {
-                    if (MemoryMarshal.TryGetArray(_preHash, out ArraySegment<byte> rentedArray))
+                    if (_preHash.Length > 0)
                     {
-                        ArrayPool<byte>.Shared.Return(rentedArray.Array!);
+                        if (MemoryMarshal.TryGetArray(_preHash, out ArraySegment<byte> rentedArray))
+                        {
+                            ArrayPool<byte>.Shared.Return(rentedArray.Array!);
+                        }
+
+                        _preHash = default;
```

Comment:
This code is used like 3 times, can we use `ClearPreHash`? Or even better change `_preHash` from `ReadOnlyMemory` to `ArraySegment` and avoid this method all together?

Reply:
Why would using ArraySegment avoid this?